### PR TITLE
chore: use docker image to test library generation scripts

### PR DESCRIPTION
**Pull request by diegomarquezp**
This PR changes `.github/workflows/verify-generation-config.yaml` to use the docker image created in https://github.com/googleapis/sdk-platform-java/pull/2493 instead of directly calling the scripts from the host.

### CODE REVIEW

**Comment by diegomarquezp** [comment]
Testing in progress
https://github.com/googleapis/google-cloud-java/actions/runs/8033790367/job/21944638790?pr=10435

**Comment by diegomarquezp** [comment]
Test was OK

File: .github/workflows/verify-generation-config.yaml [hunk 1]
```diff
@@ -9,35 +9,25 @@ jobs:
...
+        # be referenced by the container and its children
+        if [[ $(docker volume inspect repo) != '[]' ]]; then
+          docker volume rm repo
+        fi
+        docker volume create --name "repo" --opt "type=none" --opt "device=$(pwd)" --opt "o=bind"
     - name: generate from configuration
       shell: bash
       run: |
         git config --global user.email "github-workflow@github.com"
         git config --global user.name "Github Workflow"
```
Comment (JoeWang1127) [inline .github/workflows/verify-generation-config.yaml:25]: Do we still need to set git user and email since they are already set in Dockerfile?

**Review by JoeWang1127 — APPROVED** [review]

File: .github/workflows/verify-generation-config.yaml (hunk 1 above)
Comment (diegomarquezp) [reply .github/workflows/verify-generation-config.yaml:25]: We don't need to. I'll remove them

**Pull request merged**